performance/src/main/java/org/hibernate/validator/performance/simple/SimpleValidation.java

```diff
@@ -67,10 +68,11 @@
 	@Threads(50)
 	@Warmup(iterations = 10)
 	@Measurement(iterations = 50)
-	public void testSimpleBeanValidation(ValidationState state) {
+	public void testSimpleBeanValidation(ValidationState state, Blackhole bh) {
 		DriverSetup driverSetup = new DriverSetup( state );
 		Set<ConstraintViolation<Driver>> violations = state.validator.validate( driverSetup.getDriver() );
 		assertThat( violations ).hasSize( driverSetup.getExpectedViolationCount() );
+		bh.consume( violations );
```

Comment:
Out of curiousity, how comes that the assertion on the size of `violations` on the line before isn't enough to prevent the optimization you mention?